src/terminator/mod.rs

```diff
@@ -367,31 +367,43 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
         // Only trait methods can have a Self parameter.
 
+        // Intercept some methods (even if we can find MIR for them)
+        if let ty::InstanceDef::Item(def_id) = instance.def {
+            match &self.tcx.item_path_str(def_id)[..] {
+                "std::sys::imp::fast_thread_local::register_dtor" => {
```

Comment:
> Because it evaluates the static the first time it is accessed.

Ah, I see, so the `self.ecx.globals` stores the already initialized globals. `global_item` just has the side-effect of making sure the global is in that table?

>  There is no miri support for weak linkage

There's not much miri can do, is there? Just like with C abi calls, I guess.

> it turns the static into a pointer behind the scenes.

"it" = rustc or miri?